IdentityCore/tests/integration/MSIDDefaultTokenCacheIntegrationTests.m

```diff
@@ -0,0 +1,604 @@
+    
+    MSIDAccount *account = [[MSIDAccount alloc] initWithUpn:nil
+                                                       utid:DEFAULT_TEST_UTID
+                                                        uid:DEFAULT_TEST_UID];
+    
+    // Save token
+    [_cacheAccessor saveSharedRTForAccount:account
+                              refreshToken:token
+                                   context:nil
+                                     error:nil];
+    
+    NSError *error = nil;
+    NSArray *results = [_cacheAccessor getAllSharedRTsWithClientId:[MSIDTestRequestParams v2DefaultParams].clientId
+                                                           context:nil
+                                                             error:&error];
+    
+    XCTAssertNil(error);
+    XCTAssertEqual([results count], 1);
+    XCTAssertEqualObjects(results[0], token);
+}
+
+- (void)testGetAllSharedRTsAfterSaving_whenBothATandRTinCache_andAccountWithUtidUidProvided_shouldReturnItems
+{
+    MSIDAccount *account = [[MSIDAccount alloc] initWithUpn:DEFAULT_TEST_ID_TOKEN_USERNAME
+                                                       utid:DEFAULT_TEST_UTID
+                                                        uid:DEFAULT_TEST_UID];
+    
+    // Save an access token & refresh token
+    MSIDToken *accessToken = [[MSIDToken alloc] initWithTokenResponse:[MSIDTestTokenResponse v2DefaultTokenResponse]
+                                                              request:[MSIDTestRequestParams v2DefaultParams]
+                                                            tokenType:MSIDTokenTypeAccessToken];
+
+    [_cacheAccessor saveAccessToken:accessToken account:account
+                      requestParams:[MSIDTestRequestParams v2DefaultParams]
+                            context:nil
+                              error:nil];
+    
+    
+    MSIDToken *refreshToken = [[MSIDToken alloc] initWithTokenResponse:[MSIDTestTokenResponse v2DefaultTokenResponse]
+                                                               request:[MSIDTestRequestParams v2DefaultParams]
+                                                             tokenType:MSIDTokenTypeRefreshToken];
+    
+    [_cacheAccessor saveSharedRTForAccount:account
+                              refreshToken:refreshToken
+                                   context:nil
+                                     error:nil];
+
+    // retrieve all RTs
+    NSError *error = nil;
+    
+    NSArray *results = [_cacheAccessor getAllSharedRTsWithClientId:[MSIDTestRequestParams v2DefaultParams].clientId
+                                                           context:nil
+                                                             error:&error];
+    
+    XCTAssertNil(error);
+    XCTAssertEqual([results count], 1);
+    XCTAssertEqualObjects(results[0], refreshToken);
+}
+
+#pragma mark - Remove
+
+- (void)testRemoveSharedRTForAccount_whenItemInCache_andAccountWithUidUtidProvided_shouldRemoveOnlyRTItems
```

Comment:
I think we should also add a test, where RT item is passed, but has a different refreshToken value. I'll add one for legacy cache as well.

Reply:
agreed.